### ISSUE TITLE
[Other General Issues]训练优化

### ISSUE THREAD

**Issue by lazyn1997** [PaddlePaddle/PaddleDetection#4911]
## 描述问题/Describe the bug
1.我的图片分辨率是1920×1080，使用ppyolov2进行训练，我是缩放成接近原比例还是缩放成官方教程的640×640训练效果更好呢？我看有的issue的回答下说缩放比例不会有影响。
2.我使用ppyolov2_r50vd_dcn_voc.yml配置文件进行训练，只是对数据路径进行修改，完整文件内容在下方复现中，训练过程中，loss一直在下降，有些位置出现了断崖，map出现了增高到一定值下降并波动的情况，是否和配置文件中的mixup_epoch: 350以及学习率衰减有关呢，如果要避免这个现象，我应该操作呢？
![loss](https://user-images.githubusercontent.com/34644177/146307430-2a005819-7f15-4e9f-9c58-a0cf154182a2.png)

![bbox-mAP](https://user-images.githubusercontent.com/34644177/146307444-bc4aaecd-a1de-408b-9993-ffc1b28fdeb4.png)

3.为了尽可能提高map，在保持网络结构不变的情况下，我应该对那些参数进行调整来实现我的目标？

## 复现/Reproduction
* 配置文件ppyolov2_r50vd_dcn_583e_F.yml
```
_BASE_: [
  '../datasets/voc.yml',
  '../runtime.yml',
  './_base_/ppyolov2_r50vd_dcn.yml',
  './_base_/optimizer_365e.yml',
  './_base_/ppyolov2_reader.yml',
]


### voc.yml ###
num_classes: 2

TrainDataset:
  !VOCDataSet
    dataset_dir: G:/postgraduate_data/detection/LYC_ImageSets
    anno_path: trainval.txt
    label_list: label_list.txt
    data_fields: ['image', 'gt_bbox', 'gt_class', 'difficult']

EvalDataset:
  !VOCDataSet
    dataset_dir: G:/postgraduate_data/detection/LYC_ImageSets
    anno_path: test.txt
    label_list: label_list.txt
    data_fields: ['image', 'gt_bbox', 'gt_class', 'difficult']

TestDataset:
  !ImageFolder
    anno_path: G:/postgraduate_data/detection/LYC_ImageSets/label_list.txt


### runtime.yml ###
log_iter: 10
# Model save interval
snapshot_epoch: 10
save_dir: output


### ppyolov2_r50vd_dcn.yml ###
pretrain_weights: pretrain_weights/ResNet50_vd_ssld_pretrained.pdparams
YOLOv3Head:
  anchors: [[487, 264], [459, 330], [420, 386],
            [532, 314], [404, 476], [477, 406],
            [540, 372], [463, 466], [539, 472]]
weights: output/ppyolov2_r50vd_dcn_F/model_final


### ppyolov2_reader.yml ###
worker_num: 8
TrainReader:
  mixup_epoch: 350
  batch_size: 4

EvalReader:
  batch_size: 4
  collate_batch: false


### optimizer_365e.yml ###
# Total training rounds
epoch: 583

LearningRate:
  # 8gpu bs12 0.00333, 1gpu bs1 0.0000346875
  base_lr: 0.00013875
  # Learning rate adjustment strategy
  schedulers:
  - !PiecewiseDecay
    gamma: 0.1
    # Learning rate change position (rounds)
    milestones:
    - 466
    - 516
  - !LinearWarmup
    start_factor: 0.
    steps: 4000

OptimizerBuilder:
  optimizer:
    momentum: 0.9
    type: Momentum
  regularizer:
    factor: 0.0005
    type: L2

```
* 数据读取文件ppyolov2_reader.yml

```none
worker_num: 2
TrainReader:
  inputs_def:
    num_max_boxes: 5
  sample_transforms:
    - Decode: {}
    - Mixup: {alpha: 1.5, beta: 1.5}
    - RandomDistort: {}
    - RandomExpand: {fill_value: [123.675, 116.28, 103.53]}
    - RandomCrop: {}
    - RandomFlip: {}
#    - AutoAugment: {autoaug_type: v1}
  batch_transforms:
    - BatchRandomResize: {target_size: [320, 352, 384, 416, 448, 480, 512, 544, 576, 608, 640, 672], random_size: True, random_interp: True, keep_ratio: False}
    - NormalizeBox: {}
    - PadBox: {num_max_boxes: 5}
    - BboxXYXY2XYWH: {}
    - NormalizeImage: {mean: [0.485, 0.456, 0.406], std: [0.229, 0.224, 0.225], is_scale: True}
    - Permute: {}
    - Gt2YoloTarget: {anchor_masks: [[6, 7, 8], [3, 4, 5], [0, 1, 2]], anchors: [[487, 264], [459, 330], [420, 386], [532, 314], [404, 476], [477, 406], [540, 372], [463, 466], [539, 472]], downsample_ratios: [32, 16, 8]}
  batch_size: 12
  shuffle: true
  drop_last: true
  mixup_epoch: 25000
  use_shared_memory: true

EvalReader:
  sample_transforms:
    - Decode: {}
    - Resize: {target_size: [640, 640], keep_ratio: False, interp: 2}
    - NormalizeImage: {mean: [0.485, 0.456, 0.406], std: [0.229, 0.224, 0.225], is_scale: True}
    - Permute: {}
  batch_size: 8

TestReader:
  inputs_def:
    image_shape: [3, 640, 640]
  sample_transforms:
    - Decode: {}
    - Resize: {target_size: [640, 640], keep_ratio: False, interp: 2}
    - NormalizeImage: {mean: [0.485, 0.456, 0.406], std: [0.229, 0.224, 0.225], is_scale: True}
    - Permute: {}
  batch_size: 1
```

**Comment by wangxinxin08** [comment]
1. 我们做实验的结果是缩放成640 * 640不会影响我们的性能，只要你在训练和测试时保持一致即可。比如，你训练时使用不保持比例训练，测试时也不保持比例训练
2. loss出现断崖式下降主要和学习率decay有关，loss断崖式下降是符合预期的，一般lr decay之后会出现这个现象，map不增反降，说明可能最终的学习率太低，导致收敛到了局部最优。
3. 你可以使用ppyolov2_r50vd_dcn_365e_coco默认的学习率策略试下,  8卡下lr=0.005，学习率只decay一次，注意设置成350(只有前350个epoch使用mixup)或者mixup_epcoh设置成365(表示一直使用mixup)，另外，由于mixup的存在，PadBox的num_max_boxes最好设置成接近数据集中最大的bbox数量的2倍，最后，可以加载ppyolov2_r50vd_dcn在coco上的预训练模型以提高模型性能

**Comment by lazyn1997** [comment]
8卡0.005学习率的batchsize是12吗？还有ppyolov2_r50vd_dcn在coco上的预训练模型是[https://paddledet.bj.bcebos.com/models/pretrained/ResNet50_vd_ssld_pretrained.pdparams](url)网址下的模型吗？我看ppyolov2_r50vd_dcn.yml配置文件中是这么写的。

**Comment by wangxinxin08** [comment]
8卡0.005的学习率的batchsize是12，coco预训练的链接是https://paddledet.bj.bcebos.com/models/ppyolov2_r50vd_dcn_365e_coco.pdparams, 默认的是resnet50vd的预训练模型

**Comment by lazyn1997** [comment]
那我如果单卡，batchsize为4的话设置成0.005/8/3=0.000208最好吗，会不会太小呢？因为我之前就是严格按照计算公式设置成0.0001375出现了上述的陷入局部最优的情况。

**Comment by wangxinxin08** [comment]
按照学习率linear rule的法则，应该设置成0.000208，不过如果你的数据集不是coco的话，可以适当调整一下
